pkgs/servers/sql/cockroachdb/generic.nix

```diff
@@ -0,0 +1,69 @@
+{ lib, stdenv, buildGoModule, fetchurl
+, cmake, xz, which, autoconf
+, ncurses6, libedit, libunwind
+, installShellFiles
+, removeReferencesTo, go
+, version, sha256
+}:
+
+let
+  darwinDeps = [ libunwind libedit ];
+  linuxDeps  = [ ncurses6 ];
+
+  buildInputs = if stdenv.isDarwin then darwinDeps else linuxDeps;
+  nativeBuildInputs = [ installShellFiles cmake xz which autoconf ];
+
+in
+buildGoModule rec {
+  pname = "cockroach";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://binaries.cockroachdb.com/cockroach-v${version}.src.tgz";
+    inherit sha256;
+  };
+  vendorSha256 = null;
+
+  NIX_CFLAGS_COMPILE = lib.optionals stdenv.cc.isGNU [ "-Wno-error=deprecated-copy" "-Wno-error=redundant-move" "-Wno-error=pessimizing-move" ];
+
+  inherit nativeBuildInputs buildInputs;
+
+  buildPhase = ''
+    runHook preBuild
+    patchShebangs .
```

Comment:
This belongs into postPatch and should list which files/directories need patching instead of everything.